test/libsolidity/util/SoltestTypes.h

```diff
@@ -311,4 +313,29 @@ struct FunctionCall
 using Builtin = std::function<std::optional<bytes>(FunctionCall const&)>;
 using SideEffectHook = std::function<std::vector<std::string>(FunctionCall const&)>;
 
+/// LogRecord
+struct LogRecord
+{
+	/// The address of the account which created the log.
+	util::h160 creator;
+	/// The data attached to the log.
+	bytes data;
+	/// The log topics.
+	std::vector<util::h256> topics;
+
+	LogRecord(util::h160 _creator, bytes _data, std::vector<util::h256> _topics)
+		: creator(_creator), data(std::move(_data)), topics(std::move(_topics)) {}
```

Comment:
```suggestion
	LogRecord(util::h160 _creator, bytes _data, std::vector<util::h256> _topics):
		creator(_creator), data(std::move(_data)), topics(std::move(_topics)) {}
```

Reply:
Why don't you move the creator? Also do you really need a constructor or would `{}`-constructing be enough?

Reply:
I was also wondering why the constructor is needed, but it seem to be needed.

Reply:
It's because of the `logs.emplace_back()` call in `ExecutionFramework::recordedLogs()`. Apparently it tries to execute `LogRecord()` rather than `LogRecord{}` and a struct does not have the former. This is supposed to change in C++20 from what I've read.

So we could leave it as is or do `logs.push_back({x, y, z ...})` instead which will allow us to remove the constructor.

> Why don't you move the creator?

I wanted to point that out originally but then noticed that `creator` is `h160`, which is `FixedHash`, which is `std::array` so there's no benefit from moving it. It being `const&` parameter would make more sense.